server/src/lib/db/mysql.ts

```diff
@@ -42,7 +42,16 @@ export default class Mysql {
             DEFAULTS.idleTimeoutMillis,
     };
 
-    this.pool  = mysql.createPool({
+    // Empty strings are evaluated to false, so if the password is an empty
+    // string, the next one in the order of priority would be used. Prevent
+    // this.
+    if (options.password === ''
+        || (!options.password && config.MYSQLPASS === '')) {
+      // An empty password is equivalent to no password.
+      myConfig.password = null;
+    }
```

Comment:
After thinking about this some more, I think what we want to do is remove line 37 above, and just calculate the password here. Perhaps something like:
```
if (options.password !== undefined) {
  myConfig.password = options.password; {
else if (config.MYSQLPASS !== undefined) {
  myConfig.password ......
}
```

This way is more readable I think.